pkg/controller/unifiedpushserver/backups.go

```diff
@@ -0,0 +1,130 @@
+package unifiedpushserver
+
+import (
+	"fmt"
+
+	pushv1alpha1 "github.com/aerogear/unifiedpush-operator/pkg/apis/push/v1alpha1"
+	batchv1 "k8s.io/api/batch/v1"
+	batchv1beta1 "k8s.io/api/batch/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func backups(ups *pushv1alpha1.UnifiedPushServer) ([]batchv1beta1.CronJob, error) {
+	cronjobs := []batchv1beta1.CronJob{}
+	for _, upsBackup := range ups.Spec.Backups {
+		cronJobLabels := labels(ups, "backup")
+		for k, v := range upsBackup.Labels {
+			cronJobLabels[k] = v
+		}
+
+		jobLabels := map[string]string{"cronjob-name": upsBackup.Name}
+		for k, v := range cronJobLabels {
+			jobLabels[k] = v
+		}
+
+		cronjobs = append(cronjobs, batchv1beta1.CronJob{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      upsBackup.Name,
+				Namespace: ups.Namespace,
+				Labels:    cronJobLabels,
+			},
+			Spec: batchv1beta1.CronJobSpec{
+				Schedule: upsBackup.Schedule,
+				JobTemplate: batchv1beta1.JobTemplateSpec{
+					Spec: batchv1.JobSpec{
+						Template: corev1.PodTemplateSpec{
+							ObjectMeta: metav1.ObjectMeta{
+								Labels: jobLabels,
+							},
+							Spec: corev1.PodSpec{
+								// This SA needs to be created beforehand
+								// https://github.com/integr8ly/backup-container-image/tree/master/templates/openshift/rbac
+								ServiceAccountName: "backupjob",
```

Comment:
Could you not use the UPS Oper service account?

Reply:
Again, this is following how it's done in the keycloak operator. I like the idea of using a different ServiceAccount here, since it's for a different purpose (that has different access requirements). For example, this ServiceAccount needs to be able to read Secrets in whatever namespaces the Secrets it needs are placed in.

Reply:
OKs.